pandas/core/frame.py

```diff
@@ -2212,6 +2212,22 @@ def insert(self, loc, column, value, allow_duplicates=False):
         self._data.insert(
             loc, column, value, allow_duplicates=allow_duplicates)
 
+    def transform(self, **kwargs):
+        """
+
+        """
+        data = self.copy()
+
+        if not len(kwargs) == 1:
```

Comment:
Hmm. I don't think this is that much of a trap and it is certainly very convenient to be able to use a single call to transform. It's enough, I think, to simply ensure that all functions are evaluated on the original untransformed dataframe.